game-app/game-core/src/test/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcherTest.java

```diff
@@ -1,15 +1,119 @@
 package games.strategy.engine.framework.startup.ui;
 
+import static games.strategy.engine.framework.CliProperties.LOBBY_GAME_COMMENTS;
 import static games.strategy.engine.framework.startup.ui.InGameLobbyWatcher.getLobbySystemProperty;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.nullValue;
-
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.framework.startup.WatcherThreadMessaging;
+import games.strategy.net.INode;
+import games.strategy.net.IServerMessenger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Optional;
 import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
+import org.triplea.http.client.lobby.game.lobby.watcher.GamePostingRequest;
+import org.triplea.http.client.lobby.game.lobby.watcher.GamePostingResponse;
+import org.triplea.http.client.web.socket.client.connections.GameToLobbyConnection;
 
 final class InGameLobbyWatcherTest {
+
+  @Test
+  public void testNewInGameLobbyWatcher_hostNotReachable() throws UnknownHostException {
+
+    // need to set LOBBY_GAME_COMMENTS system property to avoid NPE in SystemPropertyReader.gameComments()
+    System.setProperty(LOBBY_GAME_COMMENTS, "testLobbyGameComments");
+
+    IServerMessenger mockIServerMessenger = mock(IServerMessenger.class);
+    GameToLobbyConnection mockGameToLobbyConnection = mock(GameToLobbyConnection.class);
+    WatcherThreadMessaging mockWatcherThreadMessaging = mock(WatcherThreadMessaging.class);
+    InGameLobbyWatcher mockInGameLobbyWatcher = mock(InGameLobbyWatcher.class);
+    INode mockINode = mock(INode.class);
+    GamePostingResponse mockGamePostingResponse = mock(GamePostingResponse.class);
```

Comment:
Instead of redefining these mocks for every test, I'd recommend declaring them as fields instead.
There are 2 variants to achieve this:

1. Using `@Mock`
```java
import org.mockito.junit.jupiter.MockitoExtension;

@ExtendWith(MockitoExtension.class)
class InGameLobbyWatcherTest {
  // No need to initialize, mockito does the magic behind the scenes.
  @Mock
  private IServerMessenger mockIServerMessenger;
}
```

2. Using regular fields
```java
class InGameLobbyWatcherTest {
  // JUnit creates a new instance of this class for every test, so this is fine.
  private IServerMessenger mockIServerMessenger = mock(IServerMessenger.class);
}
```

Reply:
updated to use reusable mocks at class level defined with @Mock